src/lib/dialog/dialog-config.ts

```diff
@@ -92,8 +92,11 @@ export class MatDialogConfig<D = any> {
   /** Aria label to assign to the dialog element */
   ariaLabel?: string | null = null;
 
+  // Note that this is disabled by default, because while the a11y recommendations are to focus
+  // the first focusable element, doing so prevents screen readers from reading out the
+  // rest of the dialog's content.
   /** Whether the dialog should focus the first focusable element on open. */
-  autoFocus?: boolean = true;
+  autoFocus?: boolean = false;
```

Comment:
@jelbourn these changes are based on our discussion from a while back. An alternate approach we can take is to move focus to the first focusable element immediately, rather than assigning it to the dialog container in between. This allows the screen reader to read out a bit more, but still not everything (it reads out the dialog header and buttons, but it skips the content vs the behavior in master where only the button is read out). 

I opted for this approach, because it allows for all the content to be read out, even though it's not the recommended one.